src/main/java/io/fabric8/jenkins/openshiftsync/PipelineJobListener.java

```diff
@@ -263,8 +264,26 @@ private BuildConfigProjectProperty buildConfigProjectForJob(WorkflowJob job) {
 
   private void upsertBuildConfigForJob(WorkflowJob job, BuildConfigProjectProperty buildConfigProjectProperty) {
     boolean create = false;
+
+    logger.info("Finding BuildConfig for namespace: " + buildConfigProjectProperty.getNamespace() + " name: " +
+      buildConfigProjectProperty.getName());
     BuildConfig jobBuildConfig = getOpenShiftClient().buildConfigs().
             inNamespace(buildConfigProjectProperty.getNamespace()).withName(buildConfigProjectProperty.getName()).get();
+
+    logger.info("Not able to find BuildConfig for namespace: " + buildConfigProjectProperty.getNamespace() + " name: " +
+      buildConfigProjectProperty.getName());
+
+    if (jobBuildConfig == null){
+
+      logger.info("Finding BuildConfig for namespace: " + buildConfigProjectProperty.getNamespace() +
+        " with label gitRepository: " + buildConfigProjectProperty.getName());
+      jobBuildConfig = getOpenShiftClient().buildConfigs().
+        inNamespace(buildConfigProjectProperty.getNamespace())
+          .withLabels(Collections.singletonMap("gitRepository", buildConfigProjectProperty.getName())).list().getItems().get(0);
```

Comment:
add a constant for `gitRepository` in  `Constants.java`

Reply:
And though it would mean another update to https://github.com/fabric8-launcher/launcher-backend,  the label should really be `openshift.io/gitRepository`

I also would like a comment here explaining that the https://github.com/fabric8-launcher/launcher-backend tool is currently setting such labels in the builds configs it creates.

Reply:
Yes @gabemontero launcher-backend is creating build-configs for quickstarts in OSIO and here is the PR which already get merged. https://github.com/fabric8-launcher/launcher-backend/pull/401

Reply:
i will declare the constant, but changing the constant variable will need a one more PR to launcher-backend. Is this really required ?

Reply:
thanks @piyush1594 - yeah, the label names can actually be a focal point here at the openshift side of things